packages/contracts/contracts/CyberMarketplace.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import '@openzeppelin/contracts/utils/Counters.sol';
+import '@openzeppelin/contracts/utils/Address.sol';
+//import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
+import '@openzeppelin/contracts/utils/introspection/ERC165Checker.sol';
+import '@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol';
+
+bytes4 constant NFT_SIG = 0xd9b67a26;
+
+contract CyberMarketplace is ERC1155Holder {
+  using Counters for Counters.Counter;
+
+  struct Listing {
+    uint256 id;
+    address nftContract;
+    address payable seller;
+    uint256 tokenId;
+    uint256 price;
+    uint256 endTime;
+  }
+
+  event Sell(
+    uint256 id,
```

Comment:
index event props, see https://docs.soliditylang.org/en/v0.8.10/contracts.html?highlight=indexed#events

Reply:
Since we can only index up to 3 fields, is there a good stratgey for choosing the right ones? I guess addresses (from, to, seller, ...) are good candidates

Reply:
the id is the most important.
Then I would say id, price, endTime

Reply:
or only id, then with id, you can add function to get the listingById